src/Scenes/GameScene/Renderers/Effects/NextProjectionMoveEffect.ts

```diff
@@ -0,0 +1,83 @@
+import * as PIXI from 'pixi.js';
+import {Effect} from './Effect';
+import {FastPool, Pool} from 'evidently-data-structures';
+import {Protagonist} from '../../../../GameLogic/Entities/Protagonist';
+import Constants from '../../../../Core/Constants';
+import {PlayerActionUtils} from '../../../../GameLogic/Enums';
+import {Direction8Utils} from '../../../../GameLogic/Enums/Direction8';
+import {GfxConstants} from '../../../../Core/Constants/GfxConstants';
+import {TextureStore} from 'evidently-pixi';
+
+const initialAlpha = 0.7;
+const finalAlpha = 0.3;
+const slideLength = 1000;
+const effectBreak = 500;
+const cycleLength = slideLength + effectBreak;
+
+export class NextProjectionMoveEffect extends PIXI.Sprite implements Effect {
+	private static readonly _pool: Pool<NextProjectionMoveEffect>;
+
+	public static getOne(projection: Protagonist, textureStore: TextureStore): NextProjectionMoveEffect {
```

Comment:
Stylistic not sure I'm happy with a method appearing before properties.

Reply:
My personal preference is to have static props and methods to appear before instance-specific ones, so they are bundled together depending on the context in which they're used. I think I have pretty strong feelings about it, so I'd prefer to keep it that way here but I am happy to discuss it :).